Feature to share news/posts/articles

idebenone opened the pull request:
Added an option to share news on the card.

medyo commented:
@idebenone, could you please include both the title and the URL when sharing content, like this:

it would be helpful if you could show the title before sharing so that users can have some context for what they are sharing.

Thank you

idebenone commented:
Hey @medyo !
I just did a commit with title as well as source of the article also. Please check and let me know.

medyo commented:
@idebenone 
- The UI looks a bit different from other pages, Could you make it like the settings modal (with inputs and buttons), something like the `rss` section
- Add a parent class to classes like `.link ` `.copy`... to prevent potential conflicts with other components in the future
- Please include the `title` prop for the share buttons.
- The passed props to the buttons need optimisation to eliminate redundancy, such as setting the size to 32 and passing the URL as url={data.link}.

idebenone commented:
@medyo 
![image](https://github.com/medyo/hackertab.dev/assets/114807970/23ed7371-a15a-4744-8414-8533761ae88f)

new ui of the modal matching the settings modal.
removed the prop redundancy on share buttons
added parent class in css
added title prop for share buttons.

medyo commented on src/components/Elements/CardWithActions/CardItemWithActions.tsx:
```diff
@@ -21,7 +21,7 @@ export const CardItemWithActions = ({
 }: CardItemWithActionsProps) => {
   const [showModal, setShowModal] = useState(false)
-  const [link, setLink] = useState({})
+  const [data, setData] = useState({})
```
`data` is too generic, please use a name like `socialShareData` or `socialSharePayload`

medyo commented on src/features/shareModal/components/ShareModal.tsx:
```diff
@@ -26,6 +26,16 @@ type ShareModalProps = {
   data: any
 }
 
+const sharingButtons = [
+  { component: EmailShareButton, icon: FaMailBulk, title: 'Mail' },
+  { component: FacebookShareButton, icon: FaFacebookSquare, title: 'Facebook' },
+  { component: TwitterShareButton, icon: FaTwitterSquare, title: 'Twitter' },
+  { component: RedditShareButton, icon: FaRedditSquare, title: 'Reddit' },
+  { component: LinkedinShareButton, icon: FaLinkedinIn, title: 'Linkedin' },
+  { component: TelegramShareButton, icon: FaTelegram, title: 'Telegram' },
+  { component: WhatsappShareButton, icon: FaWhatsappSquare, title: 'Whatsapp' },
+]
```
Please you use colourful icons, they look much better

medyo commented on src/features/shareModal/components/ShareModal.tsx:
```diff
@@ -59,41 +69,19 @@ export const ShareModal = ({ showModal, setShowModal, data }: ShareModalProps) =
-          <FaLinkedinIn size={32} />
-        </LinkedinShareButton>
+        {sharingButtons.map(({ component: ShareButton, icon: Icon, title: Title }, index) => (
+          <ShareButton key={index} url={data.link}>
```
You forgot to pass the `title={data.title}` to the ShareButton

idebenone replied:
Hey @medyo !

I'm confused a bit, which title do you want me to pass to the button, is it the title of the social media platform or title of the article/news/post?

medyo replied:
@idebenone I meant the article's title, so that it appears on social medias like:
<img width="601" alt="image" src="https://github.com/medyo/hackertab.dev/assets/2389584/e829d5f7-21ee-498a-b48b-872ff8a0f931">

idebenone replied:
Yeah I got your point,
But I dont think its possible, given then I'm using a library to share the url and as we know most of the websites rely on Open Graph meta tags, they dont accept additional parameters in the offsite share url.
But in X(Twitter) it looks like the above image even without supplying the title prop to the button.

medyo replied:
@idebenone I'm referring to the title that's included in the prompt, not the open graph metas. Please give this code a shot.
```
<ShareButton key={index} url={data.link} title={data.title}>
  <Icon size={32} title={Title} />
</ShareButton>
```
<img width="605" alt="image" src="https://github.com/medyo/hackertab.dev/assets/2389584/17a840e2-9f42-479c-9b80-f043ec7082fb">

idebenone replied:
@medyo 
Done 👍

medyo commented:
@idebenone the `source` text(LOBSTERS) looks brighter on light mode, please use the right css var, and keep it capitalised.

Thanks
<img width="301" alt="image" src="https://github.com/medyo/hackertab.dev/assets/2389584/4e9fc6a2-12b6-4b3b-9dc3-2af1dafcfb10">

idebenone commented:
@medyo 
Sorry for the delay and I have updated the css with right variable for source text and capitalized it

medyo commented:
@idebenone I couldn't merge your branch, it has some conflicts with the `develop` branch, can you take a look?
thanks

idebenone commented:
Hey @medyo 
I just resolved the conflicts!
(I swear to god I almost had a heart attack going through it....my first time)

medyo commented:
> Hey @medyo I just resolved the conflicts! (I swear to god I almost had a heart attack going through it....my first time)

yeah, conflicts can be really draining, but you'll improve through practice.
I hope you're good.

I've given your code a test run, and it seems that the bookmark icon is no longer displaying.

<img width="369" alt="image" src="https://github.com/medyo/hackertab.dev/assets/2389584/1aef0507-3a22-4b2b-9270-8764d935bf9f">

medyo commented:
I think it's a CSS issue, @victor-duarte can you take a look at this?

idebenone commented:
@medyo 
Yeah I just noticed....I'll try to see what's the problem

idebenone commented:
.blockActionButton:nth-child(1) {
    margin-right: 40px;
}

this will push the margin of bookmark icon from right making it visible on the screen.
Should I go with this or do I give new CSS class for both Block Action Buttons (bookmark & share)?

medyo commented:
I think we need a scalable way to allow adding more than 2 buttons.
What about wrapping these buttons(share, bookmark...) with flexbox and then applying the slide/opacity animation to whole div instead of the button?

victor-duarte commented:
👋🏻 

**Note:** When enabling tab navigation an odd scroll was displaying when focusing on the CTA due to the container being outside the card and animating into the card. The animation was applied on the button instead and the container moved within the card preventing the layout shift.

My 2 cents would be to go with a simple solution that allows having the 2 buttons and if in the future a third button is needed maybe looking into alternative card designs. I am no designer but maybe something the line X or instagram.

e.g:
<img width="500" alt="image" src="https://github.com/medyo/hackertab.dev/assets/1746030/a946179d-ce68-42ba-974d-84d8abc14b59">

### Suggestion

```
.blockActionButton {
  ...
  bottom: 40px;
  ...
}

.blockActionButton + .blockActionButton {
  bottom: 8px;
}
...
.blockRow:hover .blockActionButton,
.blockActionButton:focus-visible,
.blockActionButton:focus-visible ~ .blockActionButton, /* OPTIONAL: to improve tab navigation experience (subjective) */
.blockActionButton.active
.blockActionButton.active ~ .blockActionButton /* OPTIONAL: to display share icon when an item is book marked */ {
  ...
}

```
- Hover will display the two buttons
- Tab navigation will display the buttons independently (or partially together if optional line above is added in css). See attachments.

https://github.com/medyo/hackertab.dev/assets/1746030/c96c0cd7-7e13-495e-a4ac-682daabcfd00

https://github.com/medyo/hackertab.dev/assets/1746030/331d2ab3-90b8-4000-97fc-076c719a3b55

victor-duarte commented on src/features/shareModal/components/share.css:
```diff
@@ -0,0 +1,85 @@
+    gap: 20px;
+}
+
+@media (max-width: 768px) {
```
Suggestion: if you follow a mobile first approach you could trim down a couple of lines:

```
.shareModal {
  background-color: var(--card-background-color);
  bottom: 0;
  box-shadow: 0 0 20px #00000052;
  box-sizing: border-box;
  left: 0;
  padding-bottom: 90px;
  padding: 12px 12px 90px;
  position: absolute;
  width: 100%;
}

...

@media (min-width: 768px) {
  .shareModal {
    border-radius: 10px;
    bottom: auto;
    left: 50%;
    padding-bottom: 12px;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 450px;
  }

  .shareOptions {
    gap: 20px;
  }
}

victor-duarte commented on src/features/shareModal/components/share.css:
```diff
@@ -0,0 +1,85 @@
+
+.shareLink {
+    display: flex;
+    flex-direction: row;
```
This is the default, can be removed.

victor-duarte commented:
@idebenone, @medyo  if permission are granted I can apply the changes of the comments above.

victor-duarte commented:
@idebenone, if a hand is needed please let me know.
Thanks!

idebenone commented:
> @idebenone, if a hand is needed please let me know if you are able to grant permissions to work on the PR. Thanks!

Hey @victor-duarte 
I'm not able to find the option to allow collaborators on PR. I'll try to fix as many as possible(my hands are quiet tied rn).

idebenone commented:
@victor-duarte 

added all the aria-labels and made other css related changes that you had suggested related to shareModal
The modals doesnt close on ESC press even after setting `ariaHideApp={true}`. Need to look deeper into this.

idebenone commented:
@medyo  Could you please review my last commit?

medyo commented:
@idebenone "To address issues with smaller post rows, let's switch the flex direction to 'row' with some gaps, The current implementation isn't fitting well.

<img width="372" alt="image" src="https://github.com/medyo/hackertab.dev/assets/2389584/63febbb6-dcec-4e56-92a4-d3ab21fa4236">

+ Could you remove the position: absolute from the blockActionButton? We're currently managing the animation at the parent level

idebenone commented:
![image](https://github.com/medyo/hackertab.dev/assets/114807970/7e94cdef-adf6-473d-9b60-7289573afcc6)
![image](https://github.com/medyo/hackertab.dev/assets/114807970/632380dc-792f-4b19-bfb8-3084b38116f9)

@medyo 

position: absolute removed. Pushed margin on left and bottom to give a nice lift to the buttons. Also changed the order of the elements. With bookmark button as the first element from left, if a post is saved it will look odd when the button is active.
If everything is cool, I'll commit the changes.

idebenone commented:
@medyo Done 👍.

medyo commented:
Thanks a lot @idebenone, You did great!
